zk-token-sdk/src/zk_token_elgamal/pod/auth_encryption.rs

```diff
@@ -34,6 +44,25 @@ impl fmt::Display for AeCiphertext {
     }
 }
 
+#[cfg(not(target_os = "solana"))]
+impl FromStr for AeCiphertext {
+    type Err = ParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.len() > AE_CIPHERTEXT_MAX_BASE64_LEN {
+            return Err(ParseError::WrongSize);
+        }
+        let ciphertext_vec = BASE64_STANDARD.decode(s).map_err(|_| ParseError::Invalid)?;
+        if ciphertext_vec.len() != std::mem::size_of::<AeCiphertext>() {
```

Comment:
nit: how about making this `AE_CIPHERTEXT_LEN` since that's how you're going to try to deserialize the vec later

Reply:
Yes, definitely. Done!